apps_product_creator/models/product_product.py

```diff
@@ -87,16 +87,60 @@ def _manage_product_module(self, values):
         # Check if mandatory keys (for product module) are given
         if all([k in values_keys for k in check_keys]):
             product_obj = self.env['product.template']
+            module_version = self.env['odoo.module.version']
+            product_variant = self.env['product.product']
             product = product_obj.browse(values.get(product_tmpl_key, []))
             # If the product template is used to save an Odoo module
             if self._check_related_to_module(product):
                 attribute = self._get_version_attribute(values)
-                version = self._get_version_with_attribute(
+                version_ids = self._get_version_with_attribute(
```

Comment:
Here you renamed `version` to `version_ids` but `_get_version_with_attribute` returns you a record set.
To not mix with actual integer ids, I suggest you to rename `version_ids` to `versions`